### PR TITLE
test: update e2e/pnpm_lockfiles setup to update existing lockfiles

### PR DESCRIPTION
**Pull request by jbedard**
Removing `sharp` which doesn't seem to add any test cases not covered by others.

Adding `rollup` which contains an `optionalDependencies` to demonstrate the `optionalDependencies` key within the `packages`.

### CODE REVIEW

File: npm/private/utils.bzl [hunk 1]
```diff
@@ -132,11 +132,6 @@ def _convert_v6_packages(packages):
             dependencies[dep_name] = _convert_pnpm_v6_package_name(dep_version)
         package_info["dependencies"] = dependencies
 
-        # optionalDependencies
```
Comment (jbedard) [inline npm/private/utils.bzl:135]: FWICT `optionalDependencies` are only in the `importers`, not here in the `packages`

**Review by gregmagolan — APPROVED** [review]

File: npm/private/utils.bzl (hunk 1 above)
Comment (gregmagolan) [reply npm/private/utils.bzl:135]: Hmm. I'm not sure about that since importers only deals with direct dependencies and not optional dependencies of transitives. Lets hold on this one for now.

**Review by gregmagolan — CHANGES_REQUESTED** [review]
...

File: npm/private/utils.bzl (hunk 1 above)
Comment (jbedard) [reply npm/private/utils.bzl:135]: This is the type of edge case I want in that pnpm_lockfiles e2e test...

Comment (jbedard) [reply npm/private/utils.bzl]: I've updated this PR to demonstrate this content in the lockfile, although doesn't really test it.

File: e2e/pnpm_lockfiles/lockfile-test.bzl [hunk 2]
```diff
@@ -12,11 +12,33 @@ def lockfile_test(name = "lockfile", node_modules = "node_modules"):
...
+            ":.aspect_rules_js/node_modules/@aspect-test+a@5.0.2/pkg",
+            ":.aspect_rules_js/node_modules/@aspect-test+a@5.0.2/ref",
+
+            # Direct deps with lifecycles
+            ":.aspect_rules_js/node_modules/@aspect-test+c@2.0.2/lc",
+            ":.aspect_rules_js/node_modules/@aspect-test+c@2.0.2/pkg_lc",
+
+            # TODO: differs across lockfile versions
+            # Direct deps from custom registry
+            # ":.aspect_rules_js/node_modules/@types+node@registry.npmjs.org+@types+node@16.18.11",
```
Comment (jbedard) [inline e2e/pnpm_lockfiles/lockfile-test.bzl:38]: These ones are interesting where we generate different targets because the lockfiles are formatted different. Do we care?

**Review by gregmagolan — APPROVED** [review]
